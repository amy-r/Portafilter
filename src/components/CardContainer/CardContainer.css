
.add {
  text-align: center;
  margin-top: 0;
}

.addButton img{
  margin-top: 50px;
  height: 50px;
}

.search {
  display: block;
  color: #AA9E91;
  font-family: 'Open Sans', sans-serif;
  font-size: 16px;
  font-weight: 300;
  margin: auto;
  border-style: 1px solid black;
  padding: 5px;
  margin-top: 5px;
  height: 20px;
}