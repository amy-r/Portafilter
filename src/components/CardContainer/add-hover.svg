<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 67 67"><defs><style>.cls-1{fill:#c2a995;}.cls-2{fill:#fff;}</style></defs><title>add-hover</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1"><circle class="cls-1" cx="33.5" cy="33.5" r="33.5"/><path class="cls-2" d="M34.93,31.44H51.62v3.15H34.93V51.2H31.78V34.59H15.17V31.44H31.78V14.75h3.15Z"/></g></g></svg>