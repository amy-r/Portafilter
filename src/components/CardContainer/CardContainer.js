import React, { Component }from 'react';
import { withRouter, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { Card } from '../Card/Card';
import './CardContainer.css';
import HoverImage from "react-hover-image";
import add from './add.svg';
import addhover from '../../assets/add-hover.svg';


export class CardContainer extends Component {
 
  render() {
    const roasters = this.props.roasters.map( roaster => {
      return <Card {...roaster} />
    })

    return(
      <div>
        <h1> ROASTERS </h1>
        <h3 className='add'>
            ADD NEW ROASTER
        </h3>
          <NavLink to='/form'>
            <HoverImage
              src={add}
              hoverSrc={addhover}
            />
          </NavLink>
        {roasters}
      </div>
    )
  }
}


export const mapStateToProps = state => ({
  roasters: state.roasters,
  coffees: state.coffees
})

export default withRouter(connect(mapStateToProps, null)(CardContainer))