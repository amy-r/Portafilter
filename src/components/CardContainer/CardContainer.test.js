/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import { CardContainer, mapStateToProps, mapDispatchToProps } from './CardContainer';
import { shallow } from 'enzyme';

const mockRoasters = [
  { userId: "encryptedid",
    name: "Corvus",
    location: "Denver",
    altitude: "5280",
    equipment: "drum",
    water: "calcium",
    contact: "dom",
    email: "dom@corvus.com" },
  { userId: "encryptedid",
    name: "Boxcar",
    location: "Boulder",
    altitude: "5360",
    equipment: "drum",
    water: "calcium, flouride",
    contact: "charles",
    email: "charles@boxcar.com" }
]

describe('CardContainer', () => {

  it('should render correctly', () => {

    const wrapper = shallow(<CardContainer roasters={mockRoasters} />, {disableLifecycleMethods:true})

    expect(wrapper).toMatchSnapshot();
  })

  it('should call renderedCards', () => {
    const wrapper = shallow(<CardContainer roasters={mockRoasters} />, {disableLifecycleMethods:true})

    wrapper.instance().renderedCards = jest.fn();

    expect(wrapper.instance().renderedCards).toHaveBeenCalled;
  })
})

describe('MDTP', () => {
  it('should define roaster props for the container', () => {
    const name = "Corvus";
    const location = "Denver";
    const mockStore = {roasters: [ { name, location }]}
    const expected = [{ name, location }]
    const mapped = mapStateToProps(mockStore);
    expect(mapped.roasters).toEqual(expected);
  })

  it('should define the coffee props for the container', () => {
    const roasterName = "Corvus";
    const overAllScore= "10";
    const mockStore = {coffees: [ {roasterName, overAllScore} ]};
    const expected = [ {roasterName, overAllScore} ];
    const mapped = mapStateToProps(mockStore);
    expect(mapped.coffees).toEqual(expected);
  })

  it('should call dispatch on retrievedRoasters', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.retrievedRoasters();
    expect(mockDispatch).toHaveBeenCalled;
  })

  it('should call dispatch on retrievedCoffees', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.retrievedCoffees();
    expect(mockDispatch).toHaveBeenCalled;
  })  
})