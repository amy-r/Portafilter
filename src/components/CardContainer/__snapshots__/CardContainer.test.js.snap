// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    roasters={
        Array [
            Object {
              "altitude": "5280",
              "contact": "dom",
              "email": "dom@corvus.com",
              "equipment": "drum",
              "location": "Denver",
              "name": "Corvus",
              "userId": "71yYVDRUpOO8BR5153LMNCOmrV52",
              "water": "calcium",
            },
            Object {
              "altitude": "5360",
              "contact": "charles",
              "email": "charles@boxcar.com",
              "equipment": "drum",
              "location": "Boulder",
              "name": "Boxcar",
              "userId": "71yYVDRUpOO8BR5153LMNCOmrV52",
              "water": "calcium, flouride",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           ROASTERS 
</h1>,
        <input
          className="search"
          placeholder="SEARCH"
          type="text"
/>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="addButton"
          to="/form"
>
          <HoverImage
                    className="addButton"
                    hoverSrc="add-hover.svg"
                    src="add.svg"
          />
</NavLink>,
        <h3
          className="add"
>
          ADD NEW
</h3>,
        <div
          className="card-section"
>
          <Unknown
                    altitude="5280"
                    contact="dom"
                    email="dom@corvus.com"
                    equipment="drum"
                    location="Denver"
                    name="Corvus"
                    userId="71yYVDRUpOO8BR5153LMNCOmrV52"
                    water="calcium"
          />
          <Unknown
                    altitude="5360"
                    contact="charles"
                    email="charles@boxcar.com"
                    equipment="drum"
                    location="Boulder"
                    name="Boxcar"
                    userId="71yYVDRUpOO8BR5153LMNCOmrV52"
                    water="calcium, flouride"
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ROASTERS ",
        },
        "ref": null,
        "rendered": " ROASTERS ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "search",
          "placeholder": "SEARCH",
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <HoverImage
            className="addButton"
            hoverSrc="add-hover.svg"
            src="add.svg"
/>,
          "className": "addButton",
          "to": "/form",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "className": "addButton",
            "hoverSrc": "add-hover.svg",
            "src": "add.svg",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "ADD NEW",
          "className": "add",
        },
        "ref": null,
        "rendered": "ADD NEW",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Unknown
              altitude="5280"
              contact="dom"
              email="dom@corvus.com"
              equipment="drum"
              location="Denver"
              name="Corvus"
              userId="71yYVDRUpOO8BR5153LMNCOmrV52"
              water="calcium"
/>,
            <Unknown
              altitude="5360"
              contact="charles"
              email="charles@boxcar.com"
              equipment="drum"
              location="Boulder"
              name="Boxcar"
              userId="71yYVDRUpOO8BR5153LMNCOmrV52"
              water="calcium, flouride"
/>,
          ],
          "className": "card-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "altitude": "5280",
              "contact": "dom",
              "email": "dom@corvus.com",
              "equipment": "drum",
              "location": "Denver",
              "name": "Corvus",
              "userId": "71yYVDRUpOO8BR5153LMNCOmrV52",
              "water": "calcium",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "altitude": "5360",
              "contact": "charles",
              "email": "charles@boxcar.com",
              "equipment": "drum",
              "location": "Boulder",
              "name": "Boxcar",
              "userId": "71yYVDRUpOO8BR5153LMNCOmrV52",
              "water": "calcium, flouride",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             ROASTERS 
</h1>,
          <input
            className="search"
            placeholder="SEARCH"
            type="text"
/>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="addButton"
            to="/form"
>
            <HoverImage
                        className="addButton"
                        hoverSrc="add-hover.svg"
                        src="add.svg"
            />
</NavLink>,
          <h3
            className="add"
>
            ADD NEW
</h3>,
          <div
            className="card-section"
>
            <Unknown
                        altitude="5280"
                        contact="dom"
                        email="dom@corvus.com"
                        equipment="drum"
                        location="Denver"
                        name="Corvus"
                        userId="71yYVDRUpOO8BR5153LMNCOmrV52"
                        water="calcium"
            />
            <Unknown
                        altitude="5360"
                        contact="charles"
                        email="charles@boxcar.com"
                        equipment="drum"
                        location="Boulder"
                        name="Boxcar"
                        userId="71yYVDRUpOO8BR5153LMNCOmrV52"
                        water="calcium, flouride"
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ROASTERS ",
          },
          "ref": null,
          "rendered": " ROASTERS ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "search",
            "placeholder": "SEARCH",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <HoverImage
              className="addButton"
              hoverSrc="add-hover.svg"
              src="add.svg"
/>,
            "className": "addButton",
            "to": "/form",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "addButton",
              "hoverSrc": "add-hover.svg",
              "src": "add.svg",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "ADD NEW",
            "className": "add",
          },
          "ref": null,
          "rendered": "ADD NEW",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Unknown
                altitude="5280"
                contact="dom"
                email="dom@corvus.com"
                equipment="drum"
                location="Denver"
                name="Corvus"
                userId="71yYVDRUpOO8BR5153LMNCOmrV52"
                water="calcium"
/>,
              <Unknown
                altitude="5360"
                contact="charles"
                email="charles@boxcar.com"
                equipment="drum"
                location="Boulder"
                name="Boxcar"
                userId="71yYVDRUpOO8BR5153LMNCOmrV52"
                water="calcium, flouride"
/>,
            ],
            "className": "card-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "altitude": "5280",
                "contact": "dom",
                "email": "dom@corvus.com",
                "equipment": "drum",
                "location": "Denver",
                "name": "Corvus",
                "userId": "71yYVDRUpOO8BR5153LMNCOmrV52",
                "water": "calcium",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "altitude": "5360",
                "contact": "charles",
                "email": "charles@boxcar.com",
                "equipment": "drum",
                "location": "Boulder",
                "name": "Boxcar",
                "userId": "71yYVDRUpOO8BR5153LMNCOmrV52",
                "water": "calcium, flouride",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
